pages/studentPortal/quizcomponent/LangQuizRoute.tsx

```diff
@@ -11,91 +11,177 @@ export enum Stage {
+      { name: "Learn an in-demand skill" },
+      { name: "Work in tech" },
+      { name: "Build programs for fun" },
+      { name: "I run a startup" },
+      { name: "Become a professional developer" },
+      { name: "Communicate better with technical coworkers" },
+      { name: "Start a side hustle" },
     ],
     [
-      "Front end development",
-      "Back end development",
-      "Game development",
-      "Mobile development",
-      "Data analytics",
-      "Product management",
-      "Digital marketing",
-      "UX/UI design",
-      "I don't know / Not sure yet",
+      { name: "Front end development", result: "JavaScript", weight: 1 },
+      { name: "Back end development", result: "JavaScript", weight: 1 },
+      { name: "Game development", result: "JavaScript", weight: 1 },
+      { name: "Mobile development", result: "JavaScript", weight: 1 },
+      { name: "Data analytics", result: "Python", weight: 1 },
+      { name: "Product management", result: "JavaScript", weight: 1 },
+      { name: "Digital marketing", result: "JavaScript", weight: 1 },
+      { name: "UX/UI design", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
     ],
     [
-      "Websites",
-      "Storefront",
-      "Mobile apps",
-      "Games",
-      "Email",
-      "Tools to automate your life",
-      "I don't know / Not sure yet",
+      { name: "Websites", result: "JavaScript", weight: 1 },
+      { name: "Storefront", result: "HTML/CSS", weight: 1 },
+      { name: "Mobile apps", result: "JavaScript", weight: 1 },
+      { name: "Games", result: "JavaScript", weight: 1 },
+      { name: "Email", result: "HTML/CSS", weight: 1 },
+      { name: "Tools to automate your life", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
     ],
   ];
-  // create results state object that
-  // create custom type -- based on schema type in database
+
+  type QuizContext = {
+    title: string;
+    body: string;
+  };
+
+  const context: QuizContext[] = [
+    {
+      title: "What Coding Language Should I Learn First?",
+      body: "Take this free quiz to find out what coding language you should learn first.",
+    },
+    {
+      title: "Why do you want to learn coding?",
+      body: "Select all that apply.",
+    },
+    {
+      title: "What field of work are you interested in?",
+      body: "Select 1-3 choices.",
+    },
+    {
+      title: "What are you interested in building?",
+      body: "Select all that apply.",
+    },
+  ];
+
+  const selectFieldName = quiz.map((outerItem) => {
+    return outerItem.map((innerItem) => {
+      return innerItem.name;
+    });
+  });
+
+  const [stageResponses, setStageResponses] = useState([]);
+
+  const [score, setScore] = useState({
+    JavaScript: 0,
+    HTMLCSS: 0,
+    Python: 0,
+  });
+
+  const finalScore = (userSelections: string | any[]) => {
```

Comment:
move this function out of the UI component and into the api folder, also add a test for this function, with a good number of test cases

Reply:
Sorry, do you mean pages/API?